solr/core/src/java/org/apache/solr/handler/admin/api/AliasPropertyAPI.java

```diff
@@ -0,0 +1,262 @@
+  @Inject
+  public AliasPropertyAPI(
+      CoreContainer coreContainer,
+      SolrQueryRequest solrQueryRequest,
+      SolrQueryResponse solrQueryResponse) {
+    super(coreContainer, solrQueryRequest, solrQueryResponse);
+  }
+
+  @GET
+  @PermissionName(COLL_READ_PERM)
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
+  @Operation(
+      summary = "Get properties for a collection alias.",
+      tags = {"aliases"})
+  public GetAllAliasPropertiesResponse getAllAliasProperties(
+      @Parameter(description = "Alias Name") @PathParam("aliasName") String aliasName)
+      throws Exception {
+    recordCollectionForLogAndTracing(null, solrQueryRequest);
+
+    final GetAllAliasPropertiesResponse response =
+        instantiateJerseyResponse(GetAllAliasPropertiesResponse.class);
+    final Aliases aliases = readAliasesFromZk();
+    if (aliases != null) {
+      response.properties = aliases.getCollectionAliasProperties(aliasName);
+    }
+
+    return response;
+  }
+
+  @GET
+  @Path("/{propName}")
+  @PermissionName(COLL_READ_PERM)
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
+  @Operation(
+      summary = "Get a specific property for a collection alias.",
+      tags = {"aliases"})
+  public GetAliasPropertyResponse getAliasProperty(
+      @Parameter(description = "Alias Name") @PathParam("aliasName") String aliasName,
+      @Parameter(description = "Property Name") @PathParam("propName") String propName)
+      throws Exception {
+    recordCollectionForLogAndTracing(null, solrQueryRequest);
+
+    final GetAliasPropertyResponse response =
+        instantiateJerseyResponse(GetAliasPropertyResponse.class);
+    final Aliases aliases = readAliasesFromZk();
+    if (aliases != null) {
+      response.property = propName;
+      response.value = aliases.getCollectionAliasProperties(aliasName).get(propName);
+    }
+
+    return response;
+  }
+
+  private Aliases readAliasesFromZk() throws Exception {
+    final CoreContainer coreContainer = fetchAndValidateZooKeeperAwareCoreContainer();
+    final ZkStateReader zkStateReader = coreContainer.getZkController().getZkStateReader();
+    // Make sure we have the latest alias info, since a user has explicitly invoked an alias API
+    zkStateReader.getAliasesManager().update();
+    return zkStateReader.getAliases();
+  }
+
+  @PUT
+  @PermissionName(COLL_EDIT_PERM)
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
+  @Operation(
+      summary = "Update properties for a collection alias.",
+      tags = {"aliases"})
+  public SolrJerseyResponse updateAliasProperties(
+      @Parameter(description = "Alias Name") @PathParam("aliasName") String aliasName,
+      @RequestBody(description = "Properties that need to be updated", required = true)
+          UpdateAliasPropertiesRequestBody requestBody)
+      throws Exception {
+
+    if (requestBody == null) {
```

Comment:
@gerlowskija leaving a comment for you here. I am fuzzy on how enforcing the not-null param works here. I realized the annotations with 'required=true' are just for docs, the actual null check is not performed by the framework. checked some of the other apis and I could not find details. please let me know your thoughts.

Reply:
Good question!

You're right that any constraints added to the `@RequestBody` annotation are only considered in the docs and generated OpenAPI spec.

Jersey does offer a way to validate requests based on constraint annotations, but we're not using it currently.  They call it "bean validation".  Basically you add an extra gradle dep on `org.glassfish.jersey.ext:jersey-bean-validation`, enable validation in your Jersey apps with a specific property (`jersey.config.beanValidation.enableOutputValidationErrorEntity.server`), and add any constraint annotations (like `@NotNull`) that you need.  You can even define your own annotations which is kindof cool.

I think that we'll eventually want to utilize this everywhere we can, but right now it's hard to take advantage of it due to how the v1 code explicitly calls the Jersey classes and methods internally.  The v1 code isn't using the Jersey framework, so it can't take advantage of any bean validation, so any input checking still needs to happen explicitly as you've done here.

At least, unless there's some other option I'm missing?  Maybe we should enable bean-validation for the benefit of those v2 APIs that don't have v1 counterparts (like the list-aliases you've added in this PR)?

> checked some of the other apis and I could not find details

This is probably just looseness on my part.  I'm probably not being as rigorous in input-validation as I should be 😬

Reply:
I can't tell which way you are leaning here. are you suggesting I go for the new dependency and enable it for this api, or are you ok with leaving this as is and having a different PR for enabling annotation driven input validation for all existing apis that might need it? for me it works either way.

Reply:
Sorry for being unclear.  I was attempting to convey that there _are_ mechanisms in Jersey to enforce this, but that we shouldn't use those mechanisms (yet).

What you have here is perfect for now 👍